### DISCUSSION TITLE
Does axum provide built-in client side library?

### DISCUSSION THREAD

**Discussion by McDic** [tokio-rs/axum#2773] [Q&A] [answered]
### Summary

*This might be a stupid question. If so, I am sorry.*

I would like to send some HTTPS request from server when it is triggered by another HTTPS request sent to the server.

I can see there are some examples like [using `reqwest`](https://github.com/tokio-rs/axum/blob/main/examples/reqwest-response/src/main.rs) or [using `tokio::net`](https://github.com/tokio-rs/axum/blob/main/examples/http-proxy/src/main.rs), but did not find any example uses client library built in axum.

There might be no built-in client library provided by axum, unlike [`aiohttp` in Python](https://docs.aiohttp.org/en/stable/). If so, can I safely mix `reqwest`, `tokio-tungstenite`, etc with `axum`? I don't know much details under how async works in Rust `tokio` so I have no idea.

### axum version

0.7.5

**Comment by mladedav** [comment, marked as answer]
`axum` is for building HTTP servers, it does not provide a client.

I think a closer equivalent to `aiohttp` would be `hyper` on which both `axum` and `reqwest` is built.

It's safe to use `reqwest` inside `axum` handlers, just note that some types may have the same name but actually be different (e.g. if you wanted to take `Response` from `reqwest` and return that from an `axum` handler).

**Reply by McDic** [reply]
Thanks for fast answer.
